addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+            total_input_ids = current_filters or [input_id.id for input_id in question.survey_id.user_input_ids if input_id.state != 'new']
+            result['total_inputs'] = len(total_input_ids)
+            question_input_ids = []
+            for user_input in question.user_input_line_ids:
+                if not user_input.skipped:
+                    question_input_ids.append(user_input.user_input_id.id)
+            result['answered'] = len(set(question_input_ids) & set(total_input_ids))
+            result['skipped'] = result['total_inputs'] - result['answered']
+        return result
+
+
+class SurveyPage(models.Model):
+    '''A page for a survey.
+
+    Pages are essentially containers, allowing to group questions by ordered
+    screens.
+
+    .. note::
+        A page should be deleted if the survey it belongs to is deleted. '''
+
+    _name = 'survey.page'
+    _description = 'Survey Page'
+    _rec_name = 'title'
+    _order = 'sequence,id'
+
+    # Model Fields #
+
+    title = fields.Char(string='Page Title', required=True,
+        translate=True)
+    survey_id = fields.Many2one('survey.survey', string='Survey',
+        ondelete='cascade', required=True)
+    question_ids = fields.One2many('survey.question', 'page_id',
+        string='Questions', copy=True)
+    sequence = fields.Integer(string='Page number', default=10)
+    description = fields.Html('Description',
+        help="An introductory text to your page", translate=True,
+        oldname="note")
+
+class SurveyQuestion(models.Model):
+    ''' Questions that will be asked in a survey.
+
+    Each question can have one of more suggested answers (eg. in case of
+    dropdown choices, multi-answer checkboxes, radio buttons...).'''
+    _name = 'survey.question'
+    _description = 'Survey Question'
+    _rec_name = 'question'
+    _order = 'sequence,id'
+
+    # Model fields #
+    # Question metadata
+    page_id = fields.Many2one('survey.page', string='Survey page',
+        ondelete='cascade', required=True, default=lambda self: self.env.context.get('page_id'))
+    survey_id = fields.Many2one('survey.survey', related='page_id.survey_id', string='Survey')
+    sequence = fields.Integer(default=10)
+
+    # Question
+    question = fields.Char(string='Question Name', required=True, translate=True)
+    description = fields.Html(help="Use this field to add \
+        additional explanations about your question", translate=True,
+        oldname='descriptive_text')
+
+    # Answer
+    question_type = fields.Selection([('free_text', 'Multiple Lines Text Box'),
+            ('textbox', 'Single Line Text Box'),
+            ('numerical_box', 'Numerical Value'),
+            ('datetime', 'Date and Time'),
+            ('simple_choice', 'Multiple choice: only one answer'),
+            ('multiple_choice', 'Multiple choice: multiple answers allowed'),
+            ('matrix', 'Matrix')], string='Type of Question', size=15, required=True, default='free_text')
```

Comment:
I think we should keep size attributes in selection field when we know all possible values. see comment. 
https://code.launchpad.net/~openerp-dev/openobject-server/trunk-review-charfield-size/+merge/166219/comments/375815